### PR TITLE
API endpoints

### PR DESCRIPTION
**Pull request by iantance**
Addresses: [ECI-760](https://dripcom.atlassian.net/browse/ECI-760)

I think this is ready for a look. It's branched off of @will-in-wi's  work in progress. Adds an endpoint to set account param and integration token. Adds an endpoint to return status.

### CODE REVIEW

File: Api/Settings.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?php
+namespace Drip\Connect\Api;
+use Drip\Connect\Api\SettingsInterface;
+
+class Settings implements SettingsInterface
+{
+		/** @var \Drip\Connect\Model\ConfigurationFactory */
```
Comment (will-in-wi) [inline Api/Settings.php:7]: This whole file has somewhat goofed up indentation. Could this be fixed? We really need to get a linter on this project.

File: Model/Configuration.php [hunk 2]
```diff
@@ -4,11 +4,13 @@
 
 class Configuration
 {
+		const ACCOUNT_PARAM_PATH = 'dripconnect_general/api_settings/account_param';
     const WIS_URL_PATH = 'dripconnect_general/api_settings/wis_url';
     const INTEGRATION_TOKEN = 'dripconnect_general/api_settings/integration_token';
     const MODULE_ENABLED_PATH = 'dripconnect_general/module_settings/is_enabled';
     const SALT_PATH = 'dripconnect_general/module_settings/salt';
     const LOG_SETTINGS_PATH = 'dripconnect_general/log_settings';
+		const API_TIMEOUT_PATH = 'dripconnect_general/api_settings/timeout';
```
Comment (will-in-wi) [inline Model/Configuration.php:13]: Do we need a dynamically set timeout? Seems like something we could set as a constant and not as a setting.

File: Model/Configuration.php [hunk 3]
```diff
@@ -116,4 +152,34 @@ protected function getStoreConfig($path)
...
+		 * @param string $path
+		 * @param mixed $val
+		 */
+		protected function setWebsiteConfig($path, $val)
+		{
+				$this->resourceConfig->saveConfig(
+						$path,
+						$val,
+						\Magento\Store\Model\ScopeInterface::SCOPE_WEBSITES,
+						$this->storeId
```
Comment (will-in-wi) [inline Model/Configuration.php:181]: This won't work… Store ID and website ID aren't the same, so this'll return invalid data.

Might be worth fixing in a different PR, however, where we switch over to just having everything be based on the website scope instead of store view.

File: etc/adminhtml/system.xml [hunk 4]
```diff
@@ -27,9 +27,12 @@
             </group>
             <group id="api_settings" type="text" sortOrder="10" showInDefault="1" showInWebsite="1" showInStore="1" translate="label">
                 <label>API Settings</label>
-                <field id="integration_token" translate="label" type="text" sortOrder="21" showInDefault="1" showInWebsite="0" showInStore="0">
+                <field id="integration_token" translate="label" type="text" sortOrder="21" showInDefault="1" showInWebsite="1" showInStore="1">
```
Comment (will-in-wi) [inline etc/adminhtml/system.xml:30]: I don't think we want to set this at the store level at all…
```suggestion
                <field id="integration_token" translate="label" type="text" sortOrder="21" showInDefault="1" showInWebsite="1" showInStore="0">
```

File: etc/adminhtml/system.xml [hunk 5]
```diff
@@ -27,9 +27,12 @@
...
                     <label>Drip Integration Token</label>
                 </field>
+                <field id="account_param" translate="label" type="text" sortOrder="25" showInDefault="1" showInWebsite="1" showInStore="1">
```
Comment (will-in-wi) [inline etc/adminhtml/system.xml:33]: I don't think we want this set at the store level.
```suggestion
                <field id="account_param" translate="label" type="text" sortOrder="25" showInDefault="1" showInWebsite="1" showInStore="0">
```

File: devtools_m2/cypress/integration/RestApi.feature [hunk 6]
```diff
@@ -0,0 +1,24 @@
+Feature: REST API Interactions
+
+  I want to make plugin settings available via the REST API
+
+  Scenario: Authorized request to set integration token
+    Given I am logged into the admin interface
+      And I have configured Drip to be enabled for 'main'
```
Comment (will-in-wi) [inline devtools_m2/cypress/integration/RestApi.feature:7]: I think most of these tests might be operating at a store level and giving us bad results. We might want to do a refactor PR to switch everything to work at a website level, although that could happen after this one (or before, with a rebase…)

File: Model/Configuration.php (hunk 2 above)
Comment (iantance) [reply Model/Configuration.php:13]: `setTimeout` is still getting called somewhere, so I put this back so that things would load and I could run tests. I agree we don't need it anymore.

**Comment by iantance** [comment]
@will-in-wi This one is ready for another look.

I wasn't sure how we might want to handle the `is_enabled` setting on the plugin. I set this up to enable it when the token and account_param are set, but let me know if you think it would be better to require that people enable it in the Magento UI.

File: Api/Settings.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        $this->configFactory = $configFactory;
+        $this->storeManager = $storeManager;
+        $this->productMetadata = $productMetadata;
+        $this->moduleResource = $moduleResource;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateSettings($websiteId, $accountParam, $integrationToken) {
```
Comment (will-in-wi) [inline Api/Settings.php:34]: Might want to explicitly handle the case where the website id doesn't exist. And possibly also the case where we're setting these settings on the default website instead of the specific one.

File: Api/SettingsInterface.php [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+     * POST for integration settings API
+     * @param string $websiteId
+     * @param string $accountParam
+     * @param string $integrationToken
+     * @return string
+     */
+     public function updateSettings($websiteId, $accountParam, $integrationToken);
+
+    /**
+     * POST for integration settings API
```
Comment (will-in-wi) [inline Api/SettingsInterface.php:14]: Should this be a GET? It has no side-effects.

File: Model/Configuration.php [hunk 9]
```diff
@@ -70,6 +71,11 @@ public function getIntegrationToken()
         return $this->getConfig(self::INTEGRATION_TOKEN);
     }
 
+        public function getAccountParam()
```
Comment (will-in-wi) [inline Model/Configuration.php:74]: Some indentation weirdness here.

File: Model/Configuration.php [hunk 10]
```diff
@@ -78,6 +84,11 @@ public function isEnabled()
         return (bool) $this->getConfig(self::MODULE_ENABLED_PATH);
     }
 
+	public function enable()
+	{
+		return $this->setConfig(self::MODULE_ENABLED_PATH, 1);
```
Comment (will-in-wi) [inline Model/Configuration.php:89]: Indentation weirdness here.

Edit: I'm going to stop flagging these. It looks like we have a mix of tabs and spaces all over this diff.

**Comment by iantance** [comment]
@will-in-wi Ready for another look.

I removed that `enabled` setting.
Updates and tests to handle default scope and missing website.
Cleaned up indentation (I think).
